src/pages/user-list/user-list.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useState } from 'react';
+import { UserCard } from 'common/user-card';
+import styles from './user-list.module.scss';
+
+type UserType = {
+  id: string,
+  title: string,
+  firstName: string,
+  lastName: string,
+  picture: string,
+};
+
+export const UserList = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [users, setUsers] = useState<UserType[]>([]);
+
+  const getUSersData = async () => {
+    fetch('https://dummyapi.io/data/v1/user?limit=10', {
+      headers: {
+        'app-id': '645945fa70a1298f9606d753',
+      },
+    }).then((response) => response.json())
+      .then((response) => {
+        setUsers(response.data);
+        setIsLoading(false);
+      });
+  };
+
+  useEffect(() => {
+    getUSersData();
+  }, []);
+
+  function getGender(title: string) {
+    let gender: string;
+    if (title === 'ms' || title === 'miss') {
+      gender = 'Female';
+    } else if (title === 'mr' || title === 'mrs') {
+      gender = 'Male';
+    } else {
+      gender = 'Unknown';
+    }
+    return gender;
+  }
+
+  return (
+    <div className={styles.container}>
+      {isLoading
+        ? (
+          <div className={styles.imgContainer}>
+            <img src="src/assets/icons/spinner.gif" alt="Spinner" className={styles.spinnerImg} />
+          </div>
```

Comment:
What do you think about doing this in a external component, due to we are going to use it in a lots of components?